eclipse-collections/src/main/java/org/eclipse/collections/impl/map/immutable/AbstractImmutableMap.java

```diff
@@ -171,6 +172,22 @@ public void clear()
         return map.toImmutable();
     }
 
+    @Override
+    public ImmutableMap<K, V> newWithMap(Map<? extends K, ? extends V> map)
+    {
+        UnifiedMap<K, V> unifiedMap = UnifiedMap.newMap(this);
```

Comment:
Effective Java principles state that interface should be on the left hand side. Also, this should ideally be replaced by Maps.mutable.with().